syft/source/ocimodelsource/registry_client.go

```diff
@@ -172,26 +179,23 @@ func extractGGUFLayers(manifest *v1.Manifest) []v1.Descriptor {
 }
 
 // FetchBlobRange fetches a byte range from a blob in the registry.
-// This is used to fetch only the GGUF header without downloading the entire multi-GB file.
 func (c *RegistryClient) FetchBlobRange(_ context.Context, ref name.Reference, digest v1.Hash, maxBytes int64) ([]byte, error) {
-	// Use the remote package's Layer fetching with our options
-	// Then read only the first maxBytes
 	repo := ref.Context()
 
-	// Fetch the layer (blob) using remote.Layer
 	layer, err := remote.Layer(repo.Digest(digest.String()), c.options...)
 	if err != nil {
 		return nil, fmt.Errorf("failed to fetch layer: %w", err)
 	}
 
-	// Get the compressed reader
 	reader, err := layer.Compressed()
```

Comment:
can you verify this is truly an incremental pull of the layer?

Reply:
👍 just validated that this is done via partial downloads which closes when the reader closes